sleepecg/test/test_config.py

```diff
@@ -0,0 +1,54 @@
+# Authors: Florian Hofer
+#
+# License: BSD (3-clause)
+
+"""Tests for the configuration interface."""
+
+from pathlib import Path
+
+import pytest
+
+import sleepecg.config
+from sleepecg import get_config, set_config
+
+_TESTCONFIG_PATH = '~/.sleepecg/testdata/testconfig.yml'
+
+
+@pytest.fixture(autouse=True)
+def temp_test_config():
+    """Create, use and delete a temporary user config file for testing."""
+    user_config_path_backup = sleepecg.config._USER_CONFIG_PATH
+    sleepecg.config._USER_CONFIG_PATH = Path(_TESTCONFIG_PATH).expanduser()
+
+    yield
```

Comment:
What is this? I guess it's related to the `autouse=True`, but can you quickly explain what's happening?

Reply:
It's a general syntax for pytest fixtures, the `yield` is a placeholder for the actual testcase.
From [the pytest API reference](https://docs.pytest.org/en/6.2.x/reference.html#pytest-fixture):
"Fixtures can provide their values to test functions using return or yield statements. When using yield the code block after the yield statement is executed as teardown code regardless of the test outcome, and must yield exactly once."

I'll add comments to explain :)

Reply:
pytest is like black magic. So this means that `autouse=True` is the value that's provided to the test here?

Reply:
:D
no, autouse just lets all tests in this module use this fixture by default (instead of specifying the fixture as a parameter in each function definition)

Reply:
So what happens in this specific test in the `yield` line then?

Reply:
Oh, `temp_test_config` is not a test, but a fixture... it defines the setup and cleanup steps used for the actual testcases.

Reply:
Ah, got it!